spec/Section 5 -- Validation.md

```diff
@@ -411,6 +411,7 @@ FieldsInSetCanMerge(set):
     {set} including visiting fragments and inline fragments.
   * Given each pair of members {fieldA} and {fieldB} in {fieldsForName}:
     * {SameResponseShape(fieldA, fieldB)} must be true.
+    * {SameStreamDirective(fieldA, fieldB)} must be true.
```

Comment:
`SameStreamAndDeferDirective`?

Reply:
Our proposal limits `@defer` to fragments and fragment spreads, so only `@stream` provides a new opportunity for overlapping fields to conflict.

Reply:
@robrichard did you mind elaborating why this is the case?

Reply:
@maraisr I added an explanation to the RFC document here: https://github.com/graphql/graphql-spec/pull/774

Reply:
I have concerns about this validation rule; namely I think it breaks "composability" of a GraphQL query. If you have a fragment somewhere that contains `{ friends { name } }` and another fragment that contains `{ friends @stream { name } }` then these two fragments are composible under current GraphQL rules but would not be composable once this rule is introduced.

I feel like these fragments _could_ remain composable if we don't add this rule. Since "stream" is optional, in the above case GraphQL could determine that it has to evaluate the entire list up front anyway so it could just return the same set for both.

Similarly if you had `{ friends @stream(initialCount: 1) { name } }` and `{ friends @stream(initialCount: 5) { name } }` these could both be merged into a single `{ friends @stream(initialCount: 5) { name } }` since (as @andrewingram points out [in the WG chat](https://discord.com/channels/625400653321076807/862834364718514196/893479707163652108)) the `initialCount: 5` would satisfy both.

(This consideration makes the no-stream effectively analogous to `initialCount: ∞`.)

Reply:
That said, I like the simplicity it introduces. So I'm definitely on the fence for this one.

Reply:
We did originally do something like that, but received guidance from facebook recommending a validation rule 

See discussion here: https://github.com/graphql/graphql-js/pull/2319#discussion_r362990726